src/chainlit/element.py

```diff
@@ -144,3 +146,48 @@ def before_emit(self, text_element):
         if "content" in text_element and isinstance(text_element["content"], bytes):
             text_element["content"] = text_element["content"].decode("utf-8")
         return text_element
+
+
+@dataclass
+class Pdf(Element):
+    """Useful to send a pdf (remote or local) to the UI."""
+
+    type: ElementType = "pdf"
+    url: str = None
+    path: str = None
+    content: str = None
+
+    def persist(self, client: BaseClient, for_id: str = None):
+        if not self.content and not self.url:
+            raise ValueError("Must provide content or url")
+
+        # Either upload the content or use the url
+        url = None
+        if self.content:
+            content = b64decode(urlparse.unquote(self.content))
+            url = client.upload_element(content=content)
+        else:
+            url = self.url
+
+        if url:
+            size = getattr(self, "size", None)
+            language = getattr(self, "language", None)
+            element = client.create_element(
+                name=self.name,
+                url=url,
+                type=self.type,
+                display=self.display,
+                size=size,
+                language=language,
```

Comment:
pas sur que language soit important pour un PDF. c'est plus pour du code dans le TextElement

Reply:
Ok, I wanted to keep it generic, I've removed it, we can add it later!